From a6da7318ceecab8a56faf3ebeb19a1088d0a4d44 Mon Sep 17 00:00:00 2001
From: Alvin Chang <alvinga@andestech.com>
Date: Mon, 18 Dec 2023 13:31:21 +0800
Subject: [PATCH 151/241] target/riscv: Add some missing mrvarch_cfg bits field

1. mrvarch_cfg.Smepmp should be set to 1 if Smepmp extension is enabled.
2. mrvarch_cfg.Zicbom should be set to 1 if Zicbom extension is enabled.
3. mrvarch_cfg.Zicboz should be set to 1 if Zicboz extension is enabled.
4. mrvarch_cfg.Zbk should be set to 1 if one of Zbkb/Zbkc/Zbkx
   extensions is enabled.
5. mrvarch_cfg.Zcmp should be set to 1 if zcmp extension is enabled.

Signed-off-by: Alvin Chang <alvinga@andestech.com>
---
 target/riscv/cpu.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/target/riscv/cpu.c b/target/riscv/cpu.c
index b2e8ca2ef7..d1fe43369f 100644
--- a/target/riscv/cpu.c
+++ b/target/riscv/cpu.c
@@ -1404,6 +1404,8 @@ static void andes_csr_sync_cpu_ext(CPURISCVState *env, RISCVCPU *cpu)
                             MASK_MRVARCH_CFG_ZBC, cpu->cfg.ext_zbc);
     mrvarch_cfg = set_field(mrvarch_cfg,
                             MASK_MRVARCH_CFG_ZBS, cpu->cfg.ext_zbs);
+    mrvarch_cfg = set_field(mrvarch_cfg,
+                            MASK_MRVARCH_CFG_SMEPMP, cpu->cfg.ext_smepmp);
     mrvarch_cfg = set_field(mrvarch_cfg,
                             MASK_MRVARCH_CFG_SVINVAL, cpu->cfg.ext_svinval);
     mrvarch_cfg = set_field(mrvarch_cfg,
@@ -1412,6 +1414,14 @@ static void andes_csr_sync_cpu_ext(CPURISCVState *env, RISCVCPU *cpu)
                             MASK_MRVARCH_CFG_SSCOFPMF, cpu->cfg.ext_sscofpmf);
     mrvarch_cfg = set_field(mrvarch_cfg,
                             MASK_MRVARCH_CFG_SSTC, cpu->cfg.ext_sstc);
+    mrvarch_cfg = set_field(mrvarch_cfg,
+                            MASK_MRVARCH_CFG_ZICBOM, cpu->cfg.ext_zicbom);
+    mrvarch_cfg = set_field(mrvarch_cfg,
+                            MASK_MRVARCH_CFG_ZICBOZ, cpu->cfg.ext_zicboz);
+    mrvarch_cfg = set_field(mrvarch_cfg,
+                            MASK_MRVARCH_CFG_ZBK,
+                            cpu->cfg.ext_zbkb || cpu->cfg.ext_zbkc ||
+                            cpu->cfg.ext_zbkx);
     mrvarch_cfg = set_field(mrvarch_cfg,
                             MASK_MRVARCH_CFG_ZKN, cpu->cfg.ext_zkn);
     mrvarch_cfg = set_field(mrvarch_cfg,
@@ -1435,6 +1445,8 @@ static void andes_csr_sync_cpu_ext(CPURISCVState *env, RISCVCPU *cpu)
                             MASK_MRVARCH_CFG_ZCD, cpu->cfg.ext_zcd);
     mrvarch_cfg = set_field(mrvarch_cfg,
                             MASK_MRVARCH_CFG_ZCF, cpu->cfg.ext_zcf);
+    mrvarch_cfg = set_field(mrvarch_cfg,
+                            MASK_MRVARCH_CFG_ZCMP, cpu->cfg.ext_zcmp);
     mrvarch_cfg = set_field(mrvarch_cfg,
                             MASK_MRVARCH_CFG_ZCMT, cpu->cfg.ext_zcmt);
     env->andes_csr.csrno[CSR_MRVARCH_CFG] = mrvarch_cfg;
-- 
2.34.1

